app/lib/brexit_checker/account_jwt.rb

```diff
@@ -9,14 +10,15 @@ def encode(key = ecdsa_key, algorithmn = "ES256")
 
 private
 
-  attr_reader :criteria_keys
+  attr_reader :criteria_keys, :oauth_uri
 
   def payload
```

Comment:
I'll have a go

Reply:
I also thought about maybe storing the criteria-keys in the session, but I don't want to risk having issues with going through the checker concurrently in different tabs; I guess it could be stored under a randomly generated key which is passed in this `oauth_uri`

Reply:
Right, if I select absolutely everything I get a bad gateway error trying to view the results page:
http://finder-frontend.dev.gov.uk/transition-check/questions?c%5B%5D=aero-space&c%5B%5D=automotive&c%5B%5D=electronic-machinery&c%5B%5D=install-service-repair&c%5B%5D=marine-transport&c%5B%5D=rail-manufacture&c%5B%5D=agriculture-farm&c%5B%5D=animal-ex-food&c%5B%5D=fish-inc-wholesale&c%5B%5D=forestry&c%5B%5D=vet&c%5B%5D=personal-service&c%5B%5D=legal-service&c%5B%5D=charity&c%5B%5D=voluntary&c%5B%5D=construction&c%5B%5D=environment&c%5B%5D=defence&c%5B%5D=electricity&c%5B%5D=nuclear&c%5B%5D=oil-gas-coal&c%5B%5D=renewables&c%5B%5D=media&c%5B%5D=creative&c%5B%5D=gamble&c%5B%5D=sports&c%5B%5D=culture&c%5B%5D=finance&c%5B%5D=insurance&c%5B%5D=real-estate&c%5B%5D=health&c%5B%5D=medical-tech&c%5B%5D=pharma&c%5B%5D=digital&c%5B%5D=telecoms&c%5B%5D=consumer-goods&c%5B%5D=chemical&c%5B%5D=diamond&c%5B%5D=metal&c%5B%5D=mining&c%5B%5D=non-metal-material&c%5B%5D=justice-including-prisons&c%5B%5D=public-administration&c%5B%5D=education&c%5B%5D=research&c%5B%5D=food-drink-tobacco&c%5B%5D=motor-trade&c%5B%5D=retail-wholesale-x-food-drink-motors&c%5B%5D=accommodation&c%5B%5D=restaurants-catering&c%5B%5D=tourism&c%5B%5D=air-passenger-freight&c%5B%5D=port-airports&c%5B%5D=postal-couriers&c%5B%5D=rail-passenger-freight&c%5B%5D=road-passenger-freight&c%5B%5D=warehouse-pipeline&c%5B%5D=import-from-eu&c%5B%5D=export-to-eu&c%5B%5D=haulage-goods-across-eu-borders&c%5B%5D=provide-services-do-business-in-eu&c%5B%5D=move-goods-ni&c%5B%5D=export-to-row&c%5B%5D=trade-developing&c%5B%5D=eu-domain&c%5B%5D=ip&c%5B%5D=ip-copyright&c%5B%5D=ip-trade-marks&c%5B%5D=ip-designs&c%5B%5D=ip-patents&c%5B%5D=ip-exhaustion-rights&c%5B%5D=sell-public-sector&c%5B%5D=sell-public-sector-contracts&c%5B%5D=sell-defence-contracts&c%5B%5D=eu-uk-funding&c%5B%5D=personal-eu-org&c%5B%5D=personal-eu-org-process&c%5B%5D=personal-eu-org-use&c%5B%5D=personal-eu-org-provide&c%5B%5D=employ-eu-citizens&c%5B%5D=owns-operates-business-organisation-uk&c%5B%5D=owns-operates-business-organisation&c%5B%5D=return-to-uk&c%5B%5D=move-to-eu&c%5B%5D=visiting-driving&c%5B%5D=visiting-bring-pet&c%5B%5D=visiting-uk&c%5B%5D=visiting-ie&c%5B%5D=visiting-eu&c%5B%5D=working-uk&c%5B%5D=working-ie&c%5B%5D=working-eu&c%5B%5D=studying-uk&c%5B%5D=studying-ie&c%5B%5D=studying-eu&c%5B%5D=retired&c%5B%5D=living-row&c%5B%5D=nationality-uk&page=21

I'll push a fixup commit storing the keys in the session

Reply:
Hmm, we can't store it in the session, because until a user clicks the green button they haven't seen anything about accounts cookies so they haven't consented.  We can't take "a user has expressed interest in saving their results" as consent for cookies.

Storing the data in the rails cache doesn't seem to work (at least in development mode), and feels like the wrong solution anyway.

Given that that really long URL works on www.gov.uk, it might just be fine?  Maybe we've got some nginx configuration to allow really long URLs?

I'm inclined to leave this as it is, because it seems unlikely that any actual users will need to check all of the boxes.  What do you think?

Reply:
:thinking: humm yeah, that URL is 2298 characters long!
I think that's the right approach though unless the sort the app, as it ties our fate to the checker. If long URLs become a problem they're a problem for _both_ the logged in and logged out checker.

If we have a separate solution, then it could work for one and not the other...

Reply:
Thanks for checking this!